### ISSUE TITLE
Authentication through Personal Access Token creates organisational coupling

### ISSUE THREAD

**Issue by IncandescentChrysalis** [microsoft/azure-devops-python-api#500]
Currently, the only documented way to use this library is to authenticate with a Personal Access Token.

This seems wrong, as:
* From what I know about them, Personal Access Tokens are associated to human accounts
This means an application access to an Azure DevOps instance is tightly coupled to the existence of a human person in the organisation, which means applications access will break depending on other life cycles.
* Personal Access Token also require an expiration date, meaning applications access will break regularly, forcing some extra credentials manual (human) management on top of, and separate from, the one associated with the parent account.

Is there a plan to support other kind of authentication scheme with Azure DevOps (Server)?
One could think of some OAuth2 process, for instance, allow different flows:
* H2M, much like what is achieved through PAT, without any extra manual token lifecycle management
* M2M which would allow managing applications-specific secrets, in isolation, with authorisation and grants lifecycle delegated to a potentially automated third-party

**Comment by JohnStokes228** [comment]
i would like to add that all other azure sdks im aware of are using azure-identity for auth, can this please be updated to do the same?

https://github.com/Azure/azure-sdk-for-python/issues/36643

**Comment by vs4vijay** [comment]
Yes, We need support for azure identity SDK in Azure DevOps SDK. This is what I am doing currently as workaround, but this doesn't cater for refreshing the access token, and other related use cases.

```python
credential = DefaultAzureCredential()
token = credential.get_token("499b84ac-1321-427f-aa17-267ca6975798/.default")
credential2 = BasicTokenAuthentication({"access_token": token.token})
```

CC: @nechvatalp, @tonybaloney

**Comment by vs4vijay** [comment]
Bumping this, as this approach doesn't handle refreshing the access token or other related use cases. Your assistance in adding support of Azure Identity library in devops python SDK would be greatly appreciated.

Thank you!

CC:  @apudovkin-ms, @jasinshaban, @nechvatalp, @tonybaloney
